async/logic/Async4BackendRest_DestinationAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,543 @@
+
+					String errline;
+					
+					while((errline = errbr.readLine()) != null) {
+						destinationResponseBody.append(errline);
+					}
+					
+					return GeneralErrorRes(ErrorResponseEnum.E500);
+				} catch(Exception e1) {
+					
+					e1.printStackTrace();
+				} finally {
+					if(errbr != null) {
+						try {
+							errbr.close();
+						} catch (IOException e2) {
+							
+							e2.printStackTrace();
+						}
+					}
+				}
+			} catch(Exception e) {
+				
+				e.printStackTrace();
+			} finally {
+		    	if (br != null) {
+		    		br.close();
+		    	}
+		    }
+		} catch(Exception e) {
+			
+        	e.printStackTrace();
+			
+	    	return GeneralErrorRes(ErrorResponseEnum.E500);
+	    } 
+        	    
+	    return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+	}
+
+	private Response CreateRetryMessage(String receivemessage, StringBuilder retrymessage,  StringBuilder executiontimes) {
+    
+		Responses response = new Responses();
+		
+		String transactionId = new String();
+		String destinationUri = new String();
+		String destinationHttpMethod = new String();
+		String destinatioRrequestHeader = new String();
+		String destinationRequestBody = new String();
+		String asyncId = new String();
+		String callbackUri = new String();
+		int retryTime = 1;
+		
+		try {
+			retrymessage.delete(0, retrymessage.length());
+			
+			JSONObject jsonObject = new JSONObject(receivemessage);
+	
+	
+			transactionId = jsonObject.getString("transactionid");
+			
+			destinationUri = jsonObject.getString("destinationuri");
+			
+			destinationHttpMethod = jsonObject.getString("destinationhttpmethod");
+			
+			asyncId = jsonObject.getString("asyncid");
+			
+			callbackUri = jsonObject.getString("callbackuri");
+			
+			retryTime = Integer.parseInt(jsonObject.getString("executiontimes").toString()) + 1;
+			
+	
+			retrymessage.append("{");
+			retrymessage.append("\"transactionid\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(transactionId);
+			retrymessage.append("\"");
+			retrymessage.append(",");
+			retrymessage.append("\"destinationuri\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(destinationUri);
+			retrymessage.append("\"");
+			retrymessage.append(",");
+			retrymessage.append("\"destinationhttpmethod\"");
+			retrymessage.append(":");
+			retrymessage.append("\"");
+			retrymessage.append(destinationHttpMethod);
+			retrymessage.append("\"");
+	
+			if(receivemessage.contains("destinationrequestheader")) {
+				destinatioRrequestHeader = jsonObject.getString("destinationrequestheader");
+				
+				retrymessage.append(",");
+				retrymessage.append("\"destinationrequestheader\"");
+				retrymessage.append(":");
+				retrymessage.append("\"");
+				retrymessage.append(destinatioRrequestHeader);
+				retrymessage.append("\"");
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 415:424, 427:436.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.